### DISCUSSION TITLE
Dynamic container based on location

### DISCUSSION THREAD

**Discussion by akvadrako** [gerbera/gerbera#2337] [Q&A] [answered]
How can I create a dynamic folder based on movie path (or location?). I tried this as a filter with no luck:

```
ContentURI contains "now"
```

I'm having trouble finding documentation on this format and the available fields and operators.

**Comment by KarlStraussberger** [comment, marked as answer]
You can use upnp search strings. Run `gerbera --create-config` to get an example for the config. 

The property you ask for is `path` so you may try `path startswith "/my/root/dir"`

**Reply by akvadrako** [reply]
I can't get this to work. I see that it's searching with a reasonable query but it just returns nothing:

> 2021-12-18 13:48:43.226  debug: [sqlite_database.cc:451] run(): Running: SELECT DISTINCT "c"."id", "c"."ref_id", "c"."parent_id", "c"."object_type", "c"."upnp_class", "c"."dc_title", "c"."mime_type", "c"."flags", "c"."part_number", "c"."track_number", "c"."location", "c"."last_modified", "c"."last_updated" FROM "mt_cds_object" "c" INNER JOIN "mt_metadata" "m" ON "c"."id" = "m"."item_id" INNER JOIN "grb_cds_resource" "re" ON "c"."id" = "re"."item_id" WHERE (LOWER("c"."location") LIKE LOWER('%now%')) ORDER BY "c"."last_updated" DESC LIMIT 25 OFFSET 0

There are certainly items in the database with location containing now; I can confirm that via the item detail popup.

**Reply by KarlStraussberger** [reply]
I cannot see anything wrong in your statement (do the logs after this message point to some issue).

There is only one piece that may cause problems: the `ORDER BY` clause. If your database was created long ago it may not contain values for `last_updated`.

You can either try to sort by `title` or play with your statement in some sqlite shell.

**Reply by akvadrako** [reply]
What I can see if I run the statement in the sqlite shell is that without the WHERE clause only a few rows are returned – most of my movies are not listed.

**Reply by akvadrako** [reply]
It seems there are missing the metadata section in the details pane (dc:date and dc:created) -- and I guess because of the INNER JOIN they are being excluded. What is the fix for this?

**Reply by KarlStraussberger** [reply]
If you take a look at the current code, all the `INNER` joins are replaced by `LEFT` joins. This should help with your queries.

**Reply by akvadrako** [reply]
That's great. It seems to work with a query like this: `upnp:class derivedfrom "object.item" and path contains "/now/"`.
